src/services/webdav/backend.rs

```diff
@@ -172,9 +200,18 @@ impl Builder for WebdavBuilder {
             })?
         };
 
+        let auth = match(&self.username, &self.password) {
```

Comment:
How about saving `Basic Xxxxx` directly? We can extend our support for `digest` or `bearer` more easily.

Reply:
you mean save `format!("Basic {}", general_purpose::STANDARD.encode(format!("{}:{}", username, password)))` in `WebdavBackend::authorization` directly?

Reply:
Yep! In this way, we don't need to change the logic while adding new auth methods.

Reply:
sorry I am confused, `Basic` use base64 encode while `Digest` use md5, can we save the string encode by base64 in `WebdavBackend::authorization` ?

Reply:
> sorry I am confused, `Basic` use base64 encode while `Digest` [use md5](http://www.webdav.org/specs/rfc2617.html#n-digest-access-authentication-scheme)

Oh, there are something misunderstood here.

- I want store the header value of `Authorization` we will send in `WebdavBackend::authorization`.
- HTTP header's value must valid.
- `Digest` use md5, but it's header value will be represented in a valid header value like the following:

```txt
GET /dir/index.html HTTP/1.0
Host: localhost
Authorization: Digest username="Mufasa",
                     realm="testrealm@host.com",
                     nonce="dcd98b7102dd2f0e8b11d0f600bfb0c093",
                     uri="/dir/index.html",
                     qop=auth,
                     nc=00000001,
                     cnonce="0a4f113b",
                     response="6629fae49393a05397450978507c4ef1",
                     opaque="5ccc069c403ebaf9f0171e9517f40e41"
```

---

NOTE: We don't need to add neither `digest` nor `bearer` support in this PR.

Reply:
updated. plz review it when you have time